test/extensions/filters/http/rbac/rbac_filter_fuzz_test.cc

```diff
@@ -0,0 +1,86 @@
+#include "envoy/extensions/filters/http/rbac/v3/rbac.pb.h"
+
+#include "source/common/buffer/buffer_impl.h"
+#include "source/common/network/address_impl.h"
+#include "source/extensions/filters/http/rbac/rbac_filter.h"
+
+#include "test/extensions/filters/http/common/fuzz/http_filter_fuzzer.h"
+#include "test/extensions/filters/http/rbac/rbac_filter_fuzz.pb.validate.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/mocks/http/mocks.h"
+#include "test/mocks/runtime/mocks.h"
+#include "test/mocks/server/server_factory_context.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using Envoy::Extensions::HttpFilters::RBACFilter::RoleBasedAccessControlFilter;
+using Envoy::Extensions::HttpFilters::RBACFilter::RoleBasedAccessControlFilterConfig;
+using Envoy::Extensions::HttpFilters::RBACFilter::RoleBasedAccessControlFilterConfigSharedPtr;
+using testing::WithArgs;
+
+namespace Envoy {
+namespace Extensions {
+namespace Http {
+namespace Rbac {
+
+class ReusableFilterFactory {
+public:
+  ReusableFilterFactory()
+      : addr_(std::make_shared<Network::Address::PipeInstance>("/test/test.sock")) {
+    connection_.stream_info_.downstream_connection_info_provider_->setRemoteAddress(addr_);
+    connection_.stream_info_.downstream_connection_info_provider_->setLocalAddress(addr_);
+
+    ON_CALL(callbacks_, connection())
+        .WillByDefault(testing::Return(OptRef<const Network::Connection>{connection_}));
+  }
+
+  absl::StatusOr<std::unique_ptr<RoleBasedAccessControlFilter>>
+  newFilter(const envoy::extensions::filters::http::rbac::v3::RBAC& proto_config) {
+    RoleBasedAccessControlFilterConfigSharedPtr config;
+    try {
+      config = make_shared<RoleBasedAccessControlFilterConfig>(
+          proto_config, "stats_prefix", *stats_store_.rootScope(), context_,
+          ProtobufMessage::getStrictValidationVisitor());
+    } catch (const EnvoyException& e) {
+      return absl::InvalidArgumentError(
+          absl::StrCat("EnvoyException during validation: ", e.what()));
+    }
+    auto filter = std::make_unique<RoleBasedAccessControlFilter>(std::move(config));
+    filter->setDecoderFilterCallbacks(callbacks_);
+    return filter;
+  }
+
+private:
+  Network::Address::InstanceConstSharedPtr addr_;
+  Stats::TestUtil::TestStore stats_store_;
+  NiceMock<Server::Configuration::MockServerFactoryContext> context_;
+  NiceMock<Envoy::Http::MockStreamDecoderFilterCallbacks> callbacks_;
+  NiceMock<Envoy::Network::MockConnection> connection_;
+};
+
+DEFINE_PROTO_FUZZER(const envoy::extensions::filters::http::rbac::RbacTestCase& input) {
+  try {
+    TestUtility::validate(input);
+  } catch (const EnvoyException& e) {
+    ENVOY_LOG_MISC(debug, "EnvoyException during validation: {}", e.what());
+    return;
+  }
+
+  static ReusableFilterFactory filter_factory;
```

Comment:
Please add a comment why this is static.
In general, this should be ok if the object's state isn't modified during fuzzing. If it is modified, the fuzzer might not be validating a consistent state between iterations.